src/Symfony/Component/TypeInfo/Tests/TypeResolver/StringTypeResolverTest.php

```diff
@@ -103,12 +103,12 @@ public static function resolveDataProvider(): iterable
         yield [Type::int(), DummyWithConstants::class.'::DUMMY_INT_*'];
         yield [Type::int(), DummyWithConstants::class.'::DUMMY_INT_A'];
         yield [Type::float(), DummyWithConstants::class.'::DUMMY_FLOAT_*'];
-        yield [Type::bool(), DummyWithConstants::class.'::DUMMY_TRUE_*'];
-        yield [Type::bool(), DummyWithConstants::class.'::DUMMY_FALSE_*'];
+        yield [Type::true(), DummyWithConstants::class.'::DUMMY_TRUE_*'];
+        yield [Type::false(), DummyWithConstants::class.'::DUMMY_FALSE_*'];
         yield [Type::null(), DummyWithConstants::class.'::DUMMY_NULL_*'];
-        yield [Type::array(), DummyWithConstants::class.'::DUMMY_ARRAY_*'];
+        yield [Type::array(null, Type::union(Type::int(), Type::string())), DummyWithConstants::class.'::DUMMY_ARRAY_*'];
```

Comment:
```suggestion
        yield [Type::array(key: Type::union(Type::int(), Type::string())), DummyWithConstants::class.'::DUMMY_ARRAY_*'];
```
Is a bit more explicit to me :slightly_smiling_face:

Reply:
i was done like that before, but something popped in my mind that symfony does not use named params :man_shrugging:

Reply:
named parameters are not covered by our BC promise (except in attribute constructors). And we want our tests to use our API covered by BC guarantees, as it allows to catch mistakes where we break BC (if we have to update tests, it is a good indication that something changed that might not be intended).

Reply:
Ok, then you can keep `null`!